hypershift-operator/controllers/nodepool/inplace.go

```diff
@@ -151,32 +152,37 @@ func (r *NodePoolReconciler) reconcileMachineSet(ctx context.Context,
 
 	setMachineSetReplicas(nodePool, machineSet)
 
-	// Bubble up upgrading NodePoolUpdatingVersionConditionType.
-	// TODO (alberto): differentiate with NodePoolUpdatingConfigConditionType.
-	var status corev1.ConditionStatus
-	reason := ""
-	message := ""
-	removeStatusCondition(&nodePool.Status.Conditions, hyperv1.NodePoolUpdatingVersionConditionType)
-
-	if _, ok := machineSet.Annotations[nodePoolAnnotationUpgradeInProgressTrue]; ok {
-		status = corev1.ConditionTrue
-		reason = hyperv1.NodePoolAsExpectedConditionReason
-		message = machineSet.Annotations[nodePoolAnnotationUpgradeInProgressTrue]
+	// Bubble up config change NodePoolUpdatingConfigConditionType.
+	isUpdatingConfig := isUpdatingConfig(nodePool, targetConfigHash)
+	if isUpdatingConfig {
+		setStatusCondition(&nodePool.Status.Conditions, hyperv1.NodePoolCondition{
+			Type:               hyperv1.NodePoolUpdatingConfigConditionType,
+			Status:             corev1.ConditionTrue,
+			Reason:             hyperv1.NodePoolAsExpectedConditionReason,
+			Message:            fmt.Sprintf("Updating config in progress. Target config: %s", targetConfigHash),
+			ObservedGeneration: nodePool.Generation,
+		})
+		log.Info("NodePool config is updating",
+			"current", nodePool.GetAnnotations()[nodePoolAnnotationCurrentConfig],
+			"target", targetConfigHash)
+	} else {
+		removeStatusCondition(&nodePool.Status.Conditions, hyperv1.NodePoolUpdatingConfigConditionType)
 	}
 
-	if _, ok := machineSet.Annotations[nodePoolAnnotationUpgradeInProgressFalse]; ok {
-		status = corev1.ConditionFalse
```

Comment:
This logic is necessary to report reason and message, why are we removing it?

Reply:
See https://github.com/openshift/hypershift/blob/main/control-plane-operator/hostedclusterconfigoperator/controllers/inplaceupgrader/inplaceupgrader.go#L45-L51

Reply:
This applies also for configUpdate? (not just the releaseUpdate), I think so, but just to confirm it.